### PR TITLE
Sliderule parquet s3 example

### PR DESCRIPTION
**Pull request by scottyhq**
Not sure if this will get covered during tutorial time, but wanted to document it for the hackweek!

@jomey I'm undecided on just using rendered notebooks for this event, either we disable execution or make sure the CI environment matches the cryocloud environment...

https://github.com/ICESAT-2HackWeek/ICESat-2-Hackweek-2023/blob/31af5c800dc96798f91212bec917fa1af8ffdc35/book/_config.yml#L38

### CODE REVIEW

**Comment by jomey** [comment]
> I'm undecided on just using rendered notebooks for this event, either we disable execution or make sure the CI environment matches the cryocloud environment...

I would vote to keep the current setup of committing clean notebooks and have them executed as part of the CI. Think we can replicate the setup from the CryoCloud with this repository. Also makes a good QA/QC step to ensure the notebooks work for every participant once they run their copy.

**Comment by jomey** [comment]
I assume we can use [this environment.yaml](https://github.com/CryoInTheCloud/hub-image/blob/main/environment.yml)?

**Comment by scottyhq** [comment]
> I assume we can use [this environment.yaml](https://github.com/CryoInTheCloud/hub-image/blob/main/environment.yml)?

Yes, I think that'll work, but omitting the 'Desktop' things towards the bottom of the file like QGIS, and adding the things we require for the splashpage:

https://github.com/ICESAT-2HackWeek/ICESat-2-Hackweek-2023/blob/31af5c800dc96798f91212bec917fa1af8ffdc35/conda/environment.yml#L23-L28

Would you be willing to open a separate PR for that @jomey ? Otherwise I might get to it later today

**Comment by jomey** [comment]
> Would you be willing to open a separate PR for that @jomey ? Otherwise I might get to it later today

I can take a first crack at it. I don't have access to an OS X environment, so maybe you can test that one based of what I can get to.

Agreed on skipping the desktop items. We also need the `cookiecutter` package to build the website.

**Comment by scottyhq** [comment]
Unfortunately CI will fail for code requiring AWS credentials or environment variables automatically set on CryoCloud: 

```python
# Get Temporary AWS Credentials on CryoCloud JupyterHub
client = boto3.client('sts')
# https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/sts/client/assume_role_with_web_identity.html

with open(os.environ['AWS_WEB_IDENTITY_TOKEN_FILE']) as f:
    TOKEN = f.read()
```

You could add these environment variables as repository secrets, but they are often time-limited, so updating them all the time would be annoying. I think for notebooks that require such things, we should just commit the rendered notebook and manually check (or trust :) that it runs on CryoCloud JupyterHub

**Comment by jomey** [comment]
> I think for notebooks that require such things, we should just commit the rendered notebook and manually check (or trust :) that it runs on CryoCloud JupyterHub

Sounds like a great solution.

Will you still look into [this failure](https://github.com/ICESAT-2HackWeek/ICESat-2-Hackweek-2023/actions/runs/5626123525/job/15246310024?pr=33#step:4:746). Seems unrelated, but maybe because this is the first time the tutorial section is being changed.

File: .github/workflows/qaqc.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
     # NOTE: this isn't a comprehensive spellcheck, just common typos
     - name: Spellcheck
       if: always()
-      uses: codespell-project/actions-codespell@master
+      uses: codespell-project/actions-codespell@v2
```
Comment (jomey) [inline .github/workflows/qaqc.yaml:31]: As a side note and personal practice - I am fan of splitting such commits into two. Makes it easier to transfer this to the main template without causing any conflicts.

**Comment by jomey** [comment]
Looks like the body text under [this heading](https://deploy-preview-33--icesat-2-2023.netlify.app/tutorials/sliderule/parquet-s3.html#set-area-of-interest) has an additional `[`. Or was there supposed to be a link?

File: .github/workflows/qaqc.yaml (hunk 1 above)
Comment (scottyhq) [reply .github/workflows/qaqc.yaml:31]: multi-tasking and rushing :)

File: book/_toc.yml [hunk 2]
```diff
@@ -29,7 +29,6 @@ parts:
     - file: tutorials/index
       sections:
         - file: tutorials/example/tutorial-notebook
-      sections:
```
Comment (scottyhq) [inline book/_toc.yml]: for some reason jupyter-book didn't like this... not sure why?!

**Comment by jomey** [comment]
Curious on why [this block](https://deploy-preview-33--icesat-2-2023.netlify.app/tutorials/sliderule/parquet-s3.html#run-sliderule-processing) is tracked with a `%%time`?

**Comment by scottyhq** [comment]
> Curious on why [this block](https://deploy-preview-33--icesat-2-2023.netlify.app/tutorials/sliderule/parquet-s3.html#run-sliderule-processing) is tracked with a %%time? 

I do that out of habit with sliderule processing because without `verbose=True` you don't get much feedback, and I suppose it's nice if someone is going over this notebook to know in advance it'll take ~40 seconds

**Comment by scottyhq** [comment]
Thanks for the review @jomey I added an additional sentence to each section but want to keep it brief though since other SlideRule content will cover more background info.

**Review by jomey — APPROVED** [review]

**Pull request merged**